test/conicinterface.jl

```diff
@@ -196,16 +196,157 @@ MathProgBase.loadconicproblem!(m,
 MathProgBase.optimize!(m)
 @test MathProgBase.status(m) == :Optimal
-@test_approx_eq_eps MathProgBase.getobjval(m) -sqrt(2.0) 1e-6
-@test_approx_eq_eps MathProgBase.getsolution(m)[1] 1.0 1e-6
-@test_approx_eq_eps MathProgBase.getsolution(m)[2] 1.0/sqrt(2.0) 1e-6
-@test_approx_eq_eps MathProgBase.getsolution(m)[3] 1.0/sqrt(2.0) 1e-6
+@test_approx_eq_eps MathProgBase.getobjval(m) -sqrt(2.0) tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[1] 1.0 tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[2] 1.0/sqrt(2.0) tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[3] 1.0/sqrt(2.0) tol
+if duals
+  d = MathProgBase.getconicdual(m)
+  @test_approx_eq_eps d[1] -sqrt(2.0) tol
+  @test_approx_eq_eps d[2] -sqrt(2.0) tol
+  @test_approx_eq_eps d[3] 1.0 tol
+  @test_approx_eq_eps d[4] 1.0 tol
+end
+
+end
+
+function conicEXPtest(s::MathProgBase.AbstractMathProgSolver;duals=false, tol=1e-6)
+# Problem 4 - ExpPrimal
+# min x + y + z
+#  st  y e^(x/y) <= x, y > 0 (i.e (x, y, z) are in the exponential primal cone)
+#      x == 1
+#      y == 2
+
+println("Problem 4")
+m = MathProgBase.model(s)
+MathProgBase.loadconicproblem!(m,
+[1.0, 1.0, 1.0],
+[0.0 1.0 0.0;
+ 1.0 0.0 0.0],
+[2.0, 1.0],
+[(:Zero,1:2)],
+[(:ExpPrimal, 1:3)])
+MathProgBase.optimize!(m)
+@test MathProgBase.status(m) == :Optimal
+@test_approx_eq_eps MathProgBase.getobjval(m) 6.2974 tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[1] 1.0 tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[2] 2.0 tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[3] 3.29744 tol
+
+if duals
+  # cvx_begin
+  #     variable x, y, z;
+  #     dual variable a, b, c, d;
+  #     minimize( x + y + z );
+  #     subject to
+  #         a : x == 1;
+  #         b : y == 2
+  #         { x, y, z } == exponential( 1 );
+  # cvx_end
+  # gives
+  # a = 2.6488
+  # b = 1.8243
+  d = MathProgBase.getconicdual(m)
+  @test_approx_eq_eps d[1] 1.8243 tol
+  @test_approx_eq_eps d[2] 2.6488 tol
+end
+
+# Problem 4A - ExpPrimal
+# Same as previous, except we put :ExpPrimal on constr_cones
+println("Problem 4A")
+m = MathProgBase.model(s)
+MathProgBase.loadconicproblem!(m,
+[1.0, 1.0, 1.0],
+[0.0 1.0 0.0;
+ 1.0 0.0 0.0;
+ -1.0 0.0 0.0;
+ 0.0 -1.0 0.0;
+ 0.0 0.0 -1.0],
+[2.0, 1.0, 0.0, 0.0, 0.0],
+[(:Zero,1:2), (:ExpPrimal, 3:5)],
+[(:Free, 1:3)])
+MathProgBase.optimize!(m)
+@test MathProgBase.status(m) == :Optimal
+@test_approx_eq_eps MathProgBase.getobjval(m) 6.2974 tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[1] 1.0 tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[2] 2.0 tol
+@test_approx_eq_eps MathProgBase.getsolution(m)[3] 3.29744 tol
+
+if duals
+  d = MathProgBase.getconicdual(m)
+  @test_approx_eq_eps d[1] 1.8243 tol
+  @test_approx_eq_eps d[2] 2.6488 tol
+  # TODO: How do I calculate the dual of the exp constraint and check it with a non-SCS library?
+  #
+end
+end
+
+
+function conicSDPtest(s::MathProgBase.AbstractMathProgSolver;duals=false, tol=1e-6)
+
+function is_symmetric(A::Matrix)
```

Comment:
`issym` checks for exact match
https://github.com/JuliaLang/julia/blob/master/base/sparse/sparsematrix.jl#L1899

SCS.jl enforces symmetricity through explicit equality constraints, so we're likely going to have precision issues.

I don't think the following would be a problem here, but it's another place where I ran into issues with `issym`:
```
A = [0.8608 0.3131 0.5458; 0.3131 0.8584 0.5836; 0.5458 0.5836 1.5422];
issym(A)
true
issym(A'*A*A*A')
false
```